pilot/pkg/networking/util/util.go

```diff
@@ -385,18 +385,16 @@ func cloneLocalityLbEndpoints(endpoints []*endpoint.LocalityLbEndpoints) []*endp
 // BuildConfigInfoMetadata builds core.Metadata struct containing the
 // name.namespace of the config, the type, etc.
 func BuildConfigInfoMetadata(config config.Meta) *core.Metadata {
-	metadata := &core.Metadata{
-		FilterMetadata: map[string]*pstruct.Struct{},
-	}
-	AddConfigInfoMetadata(metadata, config)
-	return metadata
+	return AddConfigInfoMetadata(nil, config)
 }
 
 // AddConfigInfoMetadata adds name.namespace of the config, the type, etc
-// to the given core.Metadata struct.
-func AddConfigInfoMetadata(metadata *core.Metadata, config config.Meta) {
+// to the given core.Metadata struct, if metadata is not initialized, build a new metadata.
+func AddConfigInfoMetadata(metadata *core.Metadata, config config.Meta) *core.Metadata {
```

Comment:
You could do pointer-to-pointer here and not return.

Reply:
I tried that out, but does not seem to be intuitive comparing to the current code.

Reply:
Yeah, I figured it. Requires some C++ mindset :)